packages/devextreme/js/__internal/grids/grid_core/column_fixing/m_column_fixing.ts

```diff
@@ -97,9 +97,19 @@ const baseFixedColumns = <T extends ModuleType<ColumnsView>>(Base: T) => class B
     const transparentColumnIndex = getTransparentColumnIndex(fixedColumns);
     const transparentColspan = fixedColumns[transparentColumnIndex].colspan;
     const columnIndices = change && change.columnIndices;
+    const rowTypes = change?.items?.map(({ rowType }) => rowType);
 
     if (columnIndices) {
-      change.columnIndices = columnIndices.map((columnIndices) => {
+      change.columnIndices = columnIndices.map((columnIndices, idx) => {
```

Comment:
Let's simplify & refactor this function logic a little to make it more readable and easy to support & understand
Because now there are a lot of duplicate/redundant conditions and some redundant logic :)

E.g:
```typescript
private _correctColumnIndicesForFixedColumns(fixedColumns, change) {
    const columnIndicesArray = change?.columnIndices;

    if (!Array.isArray(columnIndicesArray) || columnIndicesArray.length < 1) {
      return;
    }

    const transparentColumnIndex = getTransparentColumnIndex(fixedColumns);
    const transparentColspan = fixedColumns[transparentColumnIndex].colspan;
    const transparentOffset = transparentColumnIndex + transparentColspan;
    const rowTypes = change?.items?.map(({ rowType }) => rowType);

    change.columnIndices = columnIndicesArray.map((columnIndices, idx) => {
      if (!Array.isArray(columnIndices) || columnIndices.length < 1) {
        return columnIndices;
      }

      const isGroupRow = rowTypes?.[idx] === 'group';

      if (isGroupRow) {
        return [...columnIndices];
      }

      return columnIndices.reduce((result, colIdx) => {
        switch (true) {
          case colIdx < transparentColumnIndex:
            result.push(colIdx);
            break;
          case colIdx >= transparentOffset:
            result.push(colIdx - transparentColspan + 1);
            break;
          default:
            break;
        }

        return result;
      }, []);
    });
  }
```